### PR TITLE
Vendor overview

### PR DESCRIPTION
**Pull request by almoxxx**
# Type of change

- [X] New feature (non-breaking change which adds functionality)

# Description
vendor view added, for each vendor to have insight into their own basic data. Open payments still missing.

# Checklist:

- [X] I have commented my code (or ChatGPT did), particularly in hard-to-understand areas
- [X] My changes generate no new warnings, neither in my IDE nor in my browser

### CODE REVIEW

File: src/views/VendorOverview.vue [hunk 1]
```diff
@@ -1,55 +1,125 @@
...
+          </div>
+          <div class="grid grid-cols-2 place-content-between">
+            <strong> {{ $t('telephone') }}: </strong>{{ vendorMe?.Telephone }}
+          </div>
+          <div class="grid grid-cols-2 place-content-between">
+            <strong>{{ $t('location') }}: </strong>{{ vendorMe?.Location }}
+          </div>
+          <div class="grid grid-cols-2 place-content-between">
+            <strong>{{ $t('postCode') }}:</strong> {{ vendorMe?.PLZ }}
+          </div>
```
Comment (nanu-c) [inline src/views/VendorOverview.vue:32]: We don't want to show telephone and email here

File: src/views/VendorOverview.vue [hunk 2]
```diff
@@ -1,55 +1,125 @@
...
+          <h2 className="text-2xl font-bold">QR-Code</h2>
+          <div id="canvas"></div>
+          <div><strong>Url:</strong> {{ vendorMe?.UrlID }}</div>
+          <div>
+            <strong>{{ $t('IDNumber') }}</strong
+            >: {{ vendorMe?.LicenseID }}
+          </div>
+        </div>
+        <br />
+        <div class="information my-16 mt-8 pt-8 space-y-4 text-sm">
```
Comment (nanu-c) [inline src/views/VendorOverview.vue:15]: With the big space in between, the vendors don't understand that they can scroll. I think most of them will use the qr code on their Augustin license card, so we can Hide the first part behind a button.

File: src/views/VendorOverview.vue [hunk 3]
```diff
@@ -1,55 +1,125 @@
...
+          <h2 class="text-2xl font-bold text-center">{{ $t('yourData') }}</h2>
```
Comment (nanu-c) [inline src/views/VendorOverview.vue:16]: Maybe rename it into `Your Account` and reorder the fields by the relevance to the vendor. So maybe first name, then balance in bold and bigger, then lastpayout and then the rest.

File: src/views/VendorOverview.vue [hunk 4]
```diff
@@ -1,55 +1,125 @@
...
+            <strong>{{ $t('menuCredits') }}: </strong>{{ vendorMe?.Balance }}€
+          </div>
+          <div class="grid grid-cols-2 place-content-between">
+            <strong>{{ $t('lastPayout') }}: </strong>{{ vendorMe?.LastPayout }}
+          </div>
+          <div class="grid grid-cols-2 place-content-between">
+            <strong>{{ $t('lastTransactions') }}: </strong>tbd
+          </div>
+
+          <button
```
Comment (nanu-c) [inline src/views/VendorOverview.vue:43]: Maybe use the color from the settings like in the shop?

**Review by nanu-c — CHANGES_REQUESTED** [review]
Thanks for the pull request, it's on a good way, but I think it needs still some changes.

We should also change in the backend the state response between 401 `unauthorized` and 403 `forbidden`, i am going to prepare a mr for that. It's usefull, when a vendor tries to open the backoffice or the backoffice opens the vendor page without beeing a vendor.

File: src/router/index.ts [hunk 5]
```diff
@@ -274,7 +274,7 @@ const router = createRouter({
       path: '/vendoroverview',
       name: 'Vendor Overview',
       meta: {
-        layout: BackofficeDefault,
+        layout: Default,
```
[the remark this answers is not in the document]
Comment (almoxxx) [reply src/router/index.ts:278]: I did that, and hope it works as suggested

File: src/views/VendorOverview.vue (hunk 1 above)
Comment (almoxxx) [reply src/views/VendorOverview.vue]: I left them (in the new profile page) after internal discussions

File: src/views/VendorOverview.vue (hunk 2 above)
Comment (almoxxx) [reply src/views/VendorOverview.vue]: I changed the entire site structure and added 2 additional pages (QR code and profile) to be accessed through buttons

File: src/views/VendorOverview.vue (hunk 4 above)
Comment (almoxxx) [reply src/views/VendorOverview.vue]: Did that

File: src/layouts/VendorLayout.vue [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+            v-if="settStore.imgUrl"
+            alt="Augustin logo"
+            className="logo mx-auto my-1 scale-75"
+            :src="settStore.imgUrl"
+          />
+        </slot>
+      </header>
+      <div class="h-5/6 main-container px-8">
+        <slot name="main">
+          <h1>Default Content</h1>
```
Comment (nanu-c) [inline src/layouts/VendorLayout.vue:36]: Maybe remove the text here to not show it for a short moment until it's loaded

File: src/stores/vendor.ts [hunk 7]
```diff
@@ -178,7 +200,16 @@ export const vendorsStore = defineStore('vendors', {
...
+        console.error(error)
+      }
+    },
+
+    async fetchVendorMe() {
+      try {
+        const data = await getVendorMe()
+        return data.data
+      } catch (error) {
+        throw error
```
Comment (nanu-c) [inline src/stores/vendor.ts:212]: If you want to use the error in the calling function then you need to throw it again. I changed it here for you.

File: src/views/VendorOverview.vue [hunk 8]
```diff
@@ -1,66 +1,199 @@
...
+              class="list-image-none overflow-y-auto w-full h-full border-4 border-gray-200 rounded-3xl"
+            >
+              <li
+                v-for="(OpenPayment, index) in vendorMe?.OpenPayments"
+                :key="index"
+                class="flex w-full p-1 pt-2 relative"
+              >
+                <div class="flex-none grid grid-rows-1 place-content-start">
+                  <div class="pb-1">
+                    {{ $t('date') }}: {{ formatTime(OpenPayment.Timestamp) }},
```
Comment (nanu-c) [inline src/views/VendorOverview.vue:38]: Maybe format the time like `04.12.2023 - 12:33` else it's to complicated to understand i think

File: src/views/VendorViewProfil.vue [hunk 9]
```diff
@@ -0,0 +1,78 @@
...
+            </div>
+            <div class="grid grid-cols-2 place-content-between">
+              <strong>{{ $t('postCode') }}:</strong> {{ vendorMe?.PLZ }}
+            </div>
+          </div>
+          <button
+            @click="router.push('/me')"
+            class="px-2 rounded-full bg-red-600 text-white font-bold"
+          >
+            {{ $t('back') }}
```
Comment (nanu-c) [inline src/views/VendorViewProfil.vue:32]: The back button is not consistent with the other buttons from the /me page

File: src/views/backoffice/BackofficeVendorNew.vue [hunk 10]
```diff
@@ -85,73 +138,115 @@
...
                     class="appearance-none border rounded w-full py-2 px-3 text-gray-700 leading-tight focus:outline-none focus:shadow-outline"
-                    type="date" v-model="newVendor.VendorSince" id="vendorSince" />
+                    type="date"
+                    v-model="newVendor.VendorSince"
+                    id="vendorSince"
+                  />
 
-                  <label class="block text-gray-700 text-sm font-bold mb-2 pt-3" for="onlineMap">Online Map:</label>
+                  <label class="block text-gray-700 text-sm font-bold mb-2 pt-3" for="onlineMap"
+                    >Online Map:</label
```
Comment (nanu-c) [inline src/views/backoffice/BackofficeVendorNew.vue:190]: This is not translated ;) but I don't mind for this file at the moment.

File: src/views/backoffice/BackofficeVendorNew.vue [hunk 11]
```diff
@@ -211,7 +309,27 @@ const newVendor = ref<Vendor>({
...
+      Payout: 0,
+      Price: 0,
+      Quantity: 0,
+      Receiver: 0,
+      ReceiverName: '',
+      Sender: 0,
+      SenderName: '',
+      Timestamp: ''
+    }
+  ]
```
Comment (nanu-c) [inline src/views/backoffice/BackofficeVendorNew.vue:332]: the openPayments can and should be empty while creating a new vendor. so the type here is wrong.

**Review by nanu-c — CHANGES_REQUESTED** [review]

**Review by nanu-c — APPROVED** [review]

**Pull request merged**